v7/_posts/odata features/2017-12-26-06-00-JsonBatch.md

```diff
@@ -0,0 +1,330 @@
+		  },
+		  "body": {"upn": "u6@domain.com", "givenName": "Jon", "surname": "Doe"}
+		},
+		{
+		  "id": "r8",
+		  "method": "get",
+		  "url": "https://localhost:9000/$users?",
+		  "headers": {
+			"content-type": "application/json; odata.metadata=minimal; odata.streaming=true",
+			"odata-version": "4.0"
+		  },
+		  "dependsOn": ["g1", "g2", "r2"]
+		}
+	  ]
+	}
+
+{% endhighlight %}
+
+Significant attributes for the JSON batch request are as follow:
+
+* payload is in JSON format, an expansion to the original multipart/mixed format.
+* A JSON batch request has only one top level property named `requests`, whose value is an array containing multiple requests and multiple atomic groups, each of which can contain multiple requests.
+* Request headers `Content-Type` and `Accept` designates the batch request and response format to be JSON batching. For comparison, in `multipart/mime` format these two headers are` Multipart/Mixed` with optional content type parameters. These are the header values driving the batch processing while achieving ODL compatibility of two batch formats. Typical use case is that request & response are of the same format, but it should not be a limitation for the ODL Core library batching design, as long as the header values are consistent with the payload content in the request & response.
+* `Content-Length` header of the batch request specifies the content length of the JSON batch request payload.
+* Requests belonged to same atomic group are denoted by the `atomicityGroup` attribute having the same value, e.g. "g2". JSON batch format requires that all requests belonged to same group must be adjacent in the `requests` array.
+* The `dependsOn` attribute can be used to specify an array of request prerequisites of this request. Here are the important rules for `dependsOn` derived from the spec for JSON Batch semantics:
+    * If request URL references an entity from another request, id of the corresponding request must also be included;
+	* Forward referencing is not allowed;
+	* Referencing id of request that is part of a group is not allowed, with the exception that the referenced requests are in the same group of this request. 
+
+In the example above, with JSON batch format, this request specifies the following:
+* Contains two atomic groups `g1` and `g2`;
+* Request `g1-r1` needs to be executed before `g2-r6`, which needs to be before `r8`;
+* Request `r2` can be executed in arbitrary order;
+* Request `g2-r5` can be executed either before or after `g1`, but needs to be executed before `r8`. Additionally, since `g2-r5` and `g2-6r` belong to the same atomic group `g2`, execution results need to be roll-backed if either of them failed (service specific implementation).
+
+## Typical JSON Batch Request Creation using OData-Core library
+With user-defined EDM model, user can use ODL-Core library to create JSON batch request directly, similar to Multipart/Mixed batch creation. For example:
+
+{% highlight csharp %}  
```

Comment:
try replacing "{% highlight csharp %} with "~~~csharp" and {% endhighlight %} with "~~~". i.e.;

~~~csharp
 // The following code...
 MemoryStream stream = new MemoryStream();
...
~~~

Reply:
Updated.